projects/odf_transform/odf_transform/odf_parser/odf_parser.py

```diff
@@ -338,121 +456,180 @@ def standardize_odf_units(unit_string):
-            ds["latitude_precise"].attrs.update({
-                'units': 'degrees_north',
-                'standard_name': 'latitude',
-                '_FillValue': -99}
+    # Latitude (ODF uses a place holder -99 in some of their header for latitude)
+    initial_latitude = odf_header["EVENT_HEADER"].get("INITIAL_LATITUDE", -99)
+    has_latitude_timeseries = "LATD_01" in ds
+
+    if cdm_data_type in ["Profile", "TimeSeries"]:
+        # Let's define latitude first, use start latitude if available
+        if initial_latitude != -99:
+            ds.coords["latitude"] = initial_latitude
+            ds["latitude"].attrs[original_var_field] = "EVENT_HEADER:INITIAL_LATITUDE"
+        # elif has_latitude_timeseries:
+        #     ds.coords["latitude"] = ds["LATD_01"][0].values
+        #     ds["latitude"].attrs[original_var_field] = "LATD_01[0]"
+
+        # If a latitude time series is available, copy it to a preciseLat variable as suggested by ERDDAP
+        if has_latitude_timeseries:
+            ds["preciseLat"] = ds["LATD_01"].copy()
+            ds["preciseLat"].attrs[original_var_field] = "LATD_01"
+            ds["preciseLat"].attrs.update(
+                {"units": "degrees_north", "standard_name": "latitude"}
             )
-        else:
-            ds.coords["latitude"] = ds["LATD_01"]
-            ds["latitude"].attrs[original_var_field] = "LATD_01"
-    else:
-        ds.coords["latitude"] = float(odf_header["EVENT_HEADER"]["INITIAL_LATITUDE"])
-        ds["latitude"].attrs[original_var_field] = "EVENT_HEADER:INITIAL_LATITUDE"
-
-    ds['latitude'].attrs.update({'units': 'degrees_north',
-                                 'standard_name': 'latitude',
-                                 '_FillValue': -99})
-    if "LOND_01" in ds.keys():
-        if cdm_data_type in ['Profile', 'TimeSeries']:
-            ds.coords["longitude"] = ds["LOND_01"][0].values
-            ds['longitude'].attrs[original_var_field] = 'LOND_01[0]'
-            ds['longitude_precise'] = ds["LOND_01"]
-            ds["longitude_precise"].attrs[original_var_field] = "LOND_01"
-            ds["longitude_precise"].attrs.update({
-                'units': 'degrees_east',
-                'standard_name': 'longitude',
-                '_FillValue': -999}
+    elif has_latitude_timeseries:
+        ds.coords["latitude"] = ds["LATD_01"]
+        ds["latitude"].attrs[original_var_field] = "LATD_01"
+
+    # Make sure there's a latitude
+    if "latitude" not in ds:
+        raise RuntimeError("Missing Latitude input")
+
+    # Add latitude attributes
+    ds["latitude"].attrs.update(
+        {"long_name": "Latitude", "units": "degrees_north", "standard_name": "latitude"}
+    )
+
+    # Longitude (ODF uses a place holder -999 in some of their header for longitude)
+    initial_longitude = odf_header["EVENT_HEADER"].get("INITIAL_LONGITUDE", -999)
+    has_longitude_time_series = "LOND_01" in ds
+
+    if cdm_data_type in ["Profile", "TimeSeries"]:
```

Comment:
looks like you have two `if cdm_data_type in ["Profile", "TimeSeries"]:` .. can they be combined?